### PR TITLE
[Semi-Modular] Ghostrole Refactor Proof Of Concept

### PR DESCRIPTION
**Pull request by unit0016**
## About The Pull Request
(Also known as "How DS-2 Got a syndicate bomb shaped hole in the bar")
Oh god, this is a big fucking PR - and it's all subject to change, considering it's mostly a proof of concept to pitch to Xyel on how to handle offstation roles. And may be slightly spite fueled over #7058 . Maybe! Honk.

Okay, so. TL;DR. This introduces DS-2 into SSjob and makes it a legitimate alternative to the main station in all but the research department (and access/wire door combos, but I've got that one planned out already.) - Including 'department' budget and all.
Yeah, that's right. DS-2 is now treated like any other job (Save for the crew manifest, announcement console, etc). This means they can roll antagonist.

Despite the above, I'd like to make it clear that I'd prefer both stations kept in their own affairs - at least until we can find a way to split SSResearch somehow. This means antagonists should generally stick to the station they started on - And antagonizing it - and neither should be interacting at /all/ outside of lavaland. Another reminder to mining players that /technically/ you're allowed to be hostile to the other faction down there, though rule one is ever-prevalent - Don't be a dick.

With that out of the way, here's the PR Overview:

DS-2 has been rolled into SSjob, making them a "Department" - Though this is subject to change, especially if I can get someone familiar with prefscode to make them their own table of jobs. They can be affected by choice station traits, but will not spawn outside DS-2.. thankfully.

DS-2 has been given it's own, dedicated z-level - Icebox or otherwise. **I have not touched the config to remove a space level to compensate, though.. stop me if you've heard this one before: That's very much the intention.**

DS-2 Has been given bank accounts and a budget. Hail cargoslavia! ~~Or don't, because I couldn't even get the pirate pad to work for cargo. RIP.~~

DS-2's spawn text has been made **very** robust, akin to prisoner players' policy text. Irony? - It explains the rules behind researching that are (otherwise) hidden in the ghostrole guide doc, and the fact they aren't antagonists. Let's hope people actually read!

DS-2 is now affected by radiation storms, meteors.. etcetera! Though anomalies and the like don't spawn on it, ~~but if my pipedream is realized they will.~~

The jacob's ladder fix has been changed to no longer affect space ruins in broad - rather just makes sure you can't tunnel to the DS-2 z-level... though we may have to remove it outright in future. But hey, we'll cross that bridge when we come to it.

TODO:
- [x] ~~Figure out if my hunch about overflow is correct. Stop it from happening if it is.~~ Station Trait and proc seem safe.
- [ ] Roll the black market trader into the system as I intended, and remove that stupid valid-sniffer attraction in the bag.
- [x] ~~Fix the issue where ID consoles can add syndicate access right now. As funny as it is, HOPs handing out syndicate access is definitely no bueno.~~ Jobs no longer come with the ACCESS_SYNDICATE flag. Feexed.
- [x] Remove the last parts of when I was trying to make them latejoin only

Things that are (potentially) not going to make it into this PR, but are still on the TODO list:

- [x] ~~Up the amount of syndicate access flags. A fuckton. Ideally enough to match the station's - or have them literally just /be/ the station's, if I can't figure out the card console. The latter is probably more likely.~~ Transitioned to the same access flags as the station.
- [x] Cleanup for the above - make it impossible to abuse this access without going onto the NT Station.
- [x] Move the weapons auth implant to security/command only, to act as the mindshield equivalent it was always meant to be.

Things that are ABSOLUTELY not making it into this PR, but are here to explain my general plan for making a second station work:

- [ ] Get SSResearch split. This is mandatory if this project is going literally anywhere beyond this proof of concept.
- [ ] Get an announcement system variant for DS-2, and also security and medical records - which are all going to be hard.
- [ ] CARGO. Be it shuttlebound or some hacky piratepad workaround, cargo needs to exist in some form!
- [ ] Rebuild DS-2 as a station proper, be it a copypasta or original work entirely.
- [ ] Inflate the job roster to be as robust as the station's - effectively splitting the population with a bit of elbow grease.
- [ ] Introduce a config option to enable/disable DS-2 and all jobs therein. Because.. y'know, lowpop exists.
- [ ] If it comes down to it, some mechanical way to make sure assistants/operatives can't spacetide each other's stations - because breaking into an enemy station as a literal dorm-dweller is HRP, and especially when you have the gall to complain about being gat on discord afterwards! :)

## Why It's Good For The Game
This is mainly a proof of concept meant to demonstrate how offstation roles can be handled. It isn't done in the smoothest way, and /definitely/ still has some.. 'quirks' - But whether it gets merged or not, I'd like to think it could be seen as a makeshift design document on the subject.
...If you're good at reverse engineering shitcode.

## Changelog
:cl:
add: DS-2 is now rolled into SSJob, and all the wacky hijinx that entails. Watch out, you may arrive with.. a SCARF!
add: Also added framework for generic offstation bank accounts/'jobs' that aren't ghostroles - Check the Black Market Trader for an example.
del: Deprecated DS-2 Spawners outfits, and trims. Also Shuttle pilot trim.
qol: The above two locations can be latejoined or even readied up as.
fix: a few invalid objects on DS-2.
code: Added (badly coded) support for 'Spacer' "jobs" - Which act as text in the jobs menu, meant to space things apart and describe certain sections so only the doofiest will fail to understand them.
refactor: DS-2 is now it's own dedicated z-level rather than a space ruin.
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
Based af.

**Comment by ghost** [comment]
At some point, a dream occurs that DS2 is literally the same as a regular NT station, but from Syndicate.

But why do we need the second NT station but in red color/Syndicate prefix? Just for some memes/to troll DS as NaNoTrAsEn AgEnTs?

**Comment by unit0016** [comment]
> At some point, a dream occurs that DS2 is literally the same as a regular NT station, but from Syndicate.
> 
> But why do we need the second NT station but in red color/Syndicate prefix? Just for some memes/to troll DS as NaNoTrAsEn AgEnTs?

By the point we've somehow managed two stations simultaneously - gameplay will have radically shifted to accommodate them both.

**Comment by ghost** [comment]
Seems "ok" then
And there is an idea, that some *people* will use DS-2 for *only* "testing" purposes. Just imagine that some retards will explode 
DS-2 supermatter/reactor "just for fun/testing(yes, somebody tried to launch supermatter on plasma pills only)". You should think about it, i guess.

File: code/modules/modular_computers/file_system/programs/jobmanagement.dm [hunk 1]
```diff
@@ -24,14 +24,25 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
...
-		"Nanotrasen Representative",//SKYRAT EDIT END
+		"Nanotrasen Representative",
+//SKYRAT EDIT END
 		"Chief Medical Officer",
 		"Quartermaster",
-		"Shuttle Pilot")
+//SKYRAT EDIT START// - Syndicate Jobs Start Here
+		"Operative",
+		"Syndicate Staff",
+		"Researcher",
```
[the remark this answers is not in the document]
Comment (unit0016) [reply code/modules/modular_computers/file_system/programs/jobmanagement.dm:39]: Job itself isn't actually included and likely won't be for a while

**Comment by thestubborn** [comment]
ah yes

more join screen bloat
at least the CC guys have people to interact with

**Comment by unit0016** [comment]
> ah yes
> 
> more join screen bloat
> at least the CC guys have people to interact with

At least DS-2 doesn't let you break into the ghost cafe

**Comment by unit0016** [comment]
## Minus a linter I'm not going to bother fixing due to the nature of this PR, ~~and the black market trader due to complications with icebox~~, it's complete.

It's up to maintainers/xyel to decide if this is bueno or not in direction for the server, and if it is - I'll open the PR in earnest and fix everything up.
If not, PR will close and that'll be that.

Anyways, back to mapping.

**Pull request closed without merging**